mountpoint-s3-fs/src/prefetch/backpressure_controller.rs

```diff
@@ -245,14 +245,17 @@ impl BackpressureLimiter {
     ) -> Result<Option<u64>, PrefetchReadError<E>> {
         // There is already enough read window so no need to block
         if self.read_window_end_offset > offset {
-            // Check the read window incrementing queue to see there is an early request to increase read window
-            let new_read_window_offset = if let Ok(len) = self.read_window_incrementing_queue.try_recv() {
-                self.read_window_end_offset += len as u64;
-                Some(self.read_window_end_offset)
+            // Drain increment queue
+            let old_offset = self.read_window_end_offset;
+            while let Ok(len) = self.read_window_incrementing_queue.try_recv() {
```

Comment:
this looks correct with the assumption that `read_window_incrementing_queue` may contain multiple increments, but such situations looks erroneous to me

if `read_window_incrementing_queue` was instead used to receive 2 events: `DataRead(last_read_offset)`, `PartQueueStall` it would make more sense to expect multiple items in the channel here, but that may imply significant refactoring